components/ParsePOSPayment/index.tsx

```diff
@@ -29,35 +31,75 @@ interface UpdateAmount {
 
 export enum AmountUnit {
   Sat = "SAT",
-  Cent = "CENT",
+  Cent = "CENT", // TODO: eventually depreciate this for Fiat, but don't want to break existing POS links
+  Fiat = "FIAT",
+}
+
+const defaultCurrencyMetadata: Currency = {
+  id: "USD",
+  flag: "🇺🇸",
+  name: "US Dollar",
+  symbol: "$",
+  fractionDigits: 2,
+  __typename: "Currency",
 }
 
 function ParsePayment({ defaultWalletCurrency, walletId, dispatch, state }: Props) {
-  const { usdToSats, satsToUsd } = useSatPrice()
   const router = useRouter()
   const { username, amount, sats, unit, memo } = router.query
-
-  const value = usdToSats(Number(state.currentAmount)).toFixed()
-
-  const calcUsdOrCentAmount =
-    satsToUsd(Number(sats)) < 0.01 || isNaN(satsToUsd(Number(sats)))
-      ? "(less than 1 cent)"
-      : satsToUsd(Number(sats)).toFixed(2)
-
-  const valueInUSD = `$ ${
-    unit === AmountUnit.Sat ? calcUsdOrCentAmount : formatOperand(state.currentAmount)
-  }`
-  const valueInSats = `≈ ${
-    unit === AmountUnit.Sat
-      ? formatOperand(state.currentAmount)
-      : formatOperand(value.toString())
-  } sats `
+  const { display } = parseDisplayCurrency(router.query)
+  const { currencyToSats, satsToCurrency, hasLoaded } = useRealtimePrice(display)
+  const { currencyList } = useDisplayCurrency()
+  const [valueInFiat, setValueInFiat] = React.useState("0.00")
+  const [valueInSats, setValueInSats] = React.useState(0)
+  const [currentAmount, setCurrentAmount] = React.useState(state.currentAmount)
+  const [currencyMetadata, setCurrencyMetadata] = React.useState<Currency>(
+    defaultCurrencyMetadata,
+  )
+  const [numOfChanges, setNumOfChanges] = React.useState(0)
 
   const prevUnit = React.useRef(AmountUnit.Cent)
 
+  // onload
+  // set all query params on first load, even if they are not passed
+  useEffect(() => {
+    const initialUnit = unit ?? "SAT"
+    const initialAmount = safeAmount(amount, undefined, "CENTS").toString()
+    const initialSats = safeAmount(sats, undefined, "SAT").toString()
+    const initialDisplay = display ?? localStorage.getItem("display") ?? "USD"
+    const inititalQuery = { ...router.query }
```

Comment:
typo here?
```suggestion
    const initialQuery = { ...router.query }
```
Also in line 72 and 81